json-format.md

```diff
@@ -0,0 +1,210 @@
+attributes and an associated media type.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][RFC2119].
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are mapped to JSON. This
+specification does not explicitly map each attribute, but provides a generic
+mapping model that applies to all current and future CloudEvents attributes.
+
+### 2.1. Base Type System
+
+The core [CloudEvents specification][CE] defines a minimal abstract type
+system, which this mapping leans on.
+
+### 2.2. Type System Mapping
+
+The CloudEvents type system is mapped to JSON types as follows:
+
+| CloudEvents | JSON
+|--------------|-------------------------------------------------------------
+| String       | [string][JSON-String]
+| Binary       | [string][JSON-String], [Base64-encoded][base64] binary
+| URI          | [string][JSON-String]
+| Timestamp    | [string][JSON-String]
+| Map          | [JSON object][JSON-Object]
+| Object       | [JSON value][JSON-Value]
+
+### 2.3. Mapping Object-typed Attributes
+
+The CloudEvents `data` attribute is `Object`-typed, meaning that it either
+holds a `String`, or a `Binary` value, or a `Map`. `Map` entry values are
+also `Object` typed.
+
+If an implementation determines that the actual type of an `Object` is a
+`String`, the value MUST be represented as [JSON string][JSON-String]
+expression; for `Binary`, the value MUST represented as [JSON
+string][JSON-String] expression containing the [Base64][base64] encoded binary
+value; for `Map`, the value MUST be represented as a [JSON object][JSON-Object]
+expression, whereby the index fields become member names and the associated
+values become the respective member's value.
+
+### 2.4. Examples
+
+The following table shows exemplary mappings:
+
+| CloudEvents       | Type     | Exemplary JSON Value
+|--------------------|----------|-------------------------------
+| eventType          | String   | "com.example.someevent"
+| eventTypeVersion   | String   | "1.0"
+| cloudEventsVersion | String   | "0.1"
+| source             | URI      | "/mycontext"
+| eventID            | String   | "1234-1234-1234"
+| eventTime          | Timestamp| "2018-04-05T17:31:00Z"
+| contentType        | String   | "application/json"
+| extensions         | Map      | { "extA" : "vA", "extB", "vB" }
+| data               | String   | "<much wow=\"xml\"/>"
+| data               | Binary   | "Q2xvdWRFdmVudHM="
+| data               | Map      | { "objA" : "vA", "objB", "vB" }
+
+## 3. Envelope
+
+Each CloudEvents event can be wholly represented as a JSON object.
+
+Such a representation uses the media type `application/cloudevents+json`
+
+All REQUIRED and all not omitted OPTIONAL attributes in the given event
+become members of the JSON object, with the respective JSON object member
```

Comment:
s/become/MUST become/

Reply:
We should probably mention that the order of the members in the resulting JSON is not relevant. 

And, at some point (not as part of this PR) we should say whether its ok for people to include additional json properties in there outside of the `extensions` property. Normally I've seen people say its ok but then receivers just ignore unknown ones but when you have a well-defined location for extensions it becomes less clear if that's ok since you now have two spots for them.  This could be a rat-hole discussion which is why I think a follow-on PR would be better.  :-)

Reply:
agree on the "MUST become". will change. 

Order is already addressed in https://tools.ietf.org/html/rfc7159#section-1 by stating that an object is an unordered collection, but I will add the hint.

The additional properties thing is probably a topic for the F2F. I'm in the "be lenient" camp and we do have versioning that helps with deconflicting collisions, but let's hear what others say.